src/solc_0.8/defi/rules/ContributionRules.sol

```diff
@@ -0,0 +1,205 @@
+//SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.2;
+
+import {Ownable} from "@openzeppelin/contracts-0.8/access/Ownable.sol";
+import {Address} from "@openzeppelin/contracts-0.8/utils/Address.sol";
+import {SafeMathWithRequire} from "../../common/Libraries/SafeMathWithRequire.sol";
+import {IERC721} from "@openzeppelin/contracts-0.8/token/ERC721/IERC721.sol";
+import {IERC1155} from "@openzeppelin/contracts-0.8/token/ERC1155/IERC1155.sol";
+
+contract ContributionRules is Ownable {
+    using Address for address;
+
+    uint256 internal constant DECIMALS_7 = 10000000;
+    uint256 internal constant MIDPOINT_9 = 500000000;
+    uint256 internal constant NFT_FACTOR_6 = 10000;
+    uint256 internal constant NFT_CONSTANT_3 = 9000;
+    uint256 internal constant ROOT3_FACTOR = 697;
+
+    struct MultiplierRule {
+        uint256[] ids;
+        uint256[] multipliers;
+        bool balanceOf;
+        uint256 index;
+    }
+
+    mapping(IERC721 => MultiplierRule) internal _listERC721;
+    mapping(IERC1155 => MultiplierRule) internal _listERC1155;
+    IERC721[] internal _listERC721Index;
+    IERC1155[] internal _listERC1155Index;
+
+    event ERC1155MultiplierAdded(address indexed contractERC1155, uint256[] multipliers, uint256[] ids);
+    event ERC721MultiplierAdded(address indexed contractERC721, uint256[] multipliers, uint256[] ids, bool balanceOf);
+    event ERC1155MultiplierDeleted(address indexed contractERC1155);
+    event ERC721MultiplierDeleted(address indexed contractERC721);
+
+    modifier _isContract(address account) {
+        require(account.isContract(), "ContributionRules: invalid address");
+
+        _;
+    }
+
+    function computeMultiplier(address account, uint256 amountStaked) external view returns (uint256) {
+        uint256 multiplierERC721 = _multiplierBalanceOfERC721(account);
+        uint256 multiplierERC1155 = _multiplierBalanceOfERC1155(account);
+
+        return amountStaked + ((amountStaked * (multiplierERC721 + multiplierERC1155)) / 100);
```

Comment:
remove the ```/ 100```
we're passing ratio
multiplier1 = 0.1 (and not 10%)

Reply:
done -> https://github.com/thesandboxgame/sandbox-smart-contracts/commit/3eebb77ee852e2dd4fb5f5a788b3ed2ac64fb5bc